app/services/medium_providers/default.rb

```diff
@@ -32,7 +32,11 @@ def unique_id
     end
 
     def valid?
-      entity.respond_to?(:medium) && errors.empty?
+      entity.respond_to?(:medium) && (entity.image_build? || entity.pxe_build?) && errors.empty?
```

Comment:
I believe this will break katello as katello does not override `valid?`. Instead we should add an error in `validate` method that adds an error if `entity.image_build? || entity.pxe_build?` is false.
And we would need to override this method (without duplicating code) in the bootdisk implementation.

Katello code is here: https://github.com/Katello/katello/blob/master/app/services/katello/managed_content_medium_provider.rb

Reply:
Hmmm as much as I would like to use medium framework for this, I think it's not a good fit. The problem is that the medium framework won't find any medium provider if medium is nil (leading to the error medium was not set for host xyz), thus the decision on whether to validate media or not can't be done. Chicken and egg problem.

The second issue with this is that if I want to implement medium provider for bootdisk which is exactly same as the default media provider but with some changes in regard to the validation, the only way is to assume that there are three build methods and do this via `(entity.image_build? || entity.pxe_build?)` for default and `entity.bootdisk_build?` in bootdisk which is lame - not very extensive and error prone.

Therefore I think my initial idea of providing a little bit cleaner monkey patching is better than this.

Reply:
Why can't we just change the validator on the host to check if a valid medium provider is present? Wouldn't that do the trick?

Reply:
What I mean is the following:

The default medium provider gets a check (in `validate` method) to only pass if `(entity.image_build? || entity.pxe_build?)`.

Bootdisk gets a new medium provider that inherits from the default one and just overrides this check to `entity.bootdisk_build?`. If we extract that to a method overriding would be super easy.

The validator on the host object will fail if no valid medium provider is present. WDYT?